client/state/selectors/find-theme-filter-term.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { find, some } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getThemeFilters, getThemeFilterTerm } from './';
+
+/**
+ * Returns a theme filter term object that corresponds to a given filter term slug
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {String}  search The term to search for
+ * @return {Object}         A filter term object
+ */
+export default function findThemeFilterTerm( state, search ) {
```

Comment:
In the fillter lib we build a [table of terms to tax](https://github.com/Automattic/wp-calypso/blob/bbc3832a80980a0ac37379a0bb73d66acbf3806e/client/my-sites/themes/theme-filters.js#L194) (would be more data now we've added to the endpoint), that would make this a simple lookup in the table, I think?